### PR TITLE
ADF-29 | misc. settings

### PR DESCRIPTION
**Pull request by TinyTakinTeller**
image

qqlw12

### CODE REVIEW

File: scenes/ui/screen/settings_screen/settings_screen.gd [hunk 1]
```diff
@@ -23,9 +31,35 @@ func _ready() -> void:
...
+	master_settings_slider.get_title_label().text = Locale.get_ui_label("master")
+	music_settings_slider.get_title_label().text = Locale.get_ui_label("music")
+	sfx_settings_slider.get_title_label().text = Locale.get_ui_label("sfx")
+
+	shake_settings_slider.get_title_label().text = Locale.get_ui_label("shake")
+	typing_settings_slider.get_title_label().text = Locale.get_ui_label("typing")
+
+	display_mode_button.text = "?"
+	display_resolution_button.text = "?"
+	## TODO: replace resolution setting with a different one
```
Comment (TinyTakinTeller) [inline scenes/ui/screen/settings_screen/settings_screen.gd:48]: supporting "resolution" setting for this kind of game seems unnecessary right now 

i'm open to ideas for another toggle setting here

File: scenes/ui/settings_slider/settings_slider.gd [hunk 2]
```diff
@@ -2,11 +2,13 @@ class_name SettingsSlider extends MarginContainer
...
 signal data_changed(toggle: bool, value: float)
 
+@export var toggle_button_ui: bool = true
+
 var _toggle: bool
 var _value: float
 
 @onready var toggle_button: Button = %ToggleButton
-@onready var title_label: Label = %TitleLabel
+@onready var title_label: LabelTyping = %TitleLabel
```
Comment (mielifica) [inline scenes/ui/settings_slider/settings_slider.gd:11]: I wonder if it's possible to organize the scene tree of `settings_screen` something like this:

image

And then the `settings_slider` checks if there's a `Label` child and then uses that as the `title_label`
Then the shake settings can get a normal `Label` with `ShakeShaderComponent` already attached, and the typing settings can get a `LabelTyping`

(note: this is me trying to apply React reasoning to Godot :P)

File: scenes/component/control/label/label_typing/label_typing.gd [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+
+#############
+## methods ##
+#############
+
+
+func play_typing_animation(loop: bool = false) -> void:
+	_loop = loop
+	var animation_length: float = self.text.length() * _typing_delay
```
Comment (mielifica) [inline scenes/component/control/label/label_typing/label_typing.gd:47]: it's maybe a bit strange that the typing gets slower when the value gets higher (compared to the shake setting that gets more when the value is higher)

File: .github/workflows/gdlint.yml [hunk 4]
```diff
@@ -9,19 +9,27 @@ jobs:
...
     runs-on: ubuntu-latest
     
+    # workaround for locally running 'act -j gdlint' : https://github.com/nektos/act/issues/1856#issuecomment-1679427871
+    #container:
+    # image: ghcr.io/catthehacker/ubuntu:act-latest
+
+    # The allowed amount of linter problems
     env:
-      PROBLEMS_THRESHOLD: 88 # The allowed amount of linter problems
+      PROBLEMS_THRESHOLD: 0
```
Comment (mielifica) [inline .github/workflows/gdlint.yml:18]: 🎈

**Review by mielifica — APPROVED** [review]
Nice, added two remarks for consideration, but nothing super important

File: scenes/component/control/label/label_typing/label_typing.gd (hunk 3 above)
Comment (TinyTakinTeller) [reply scenes/component/control/label/label_typing/label_typing.gd:47]: I was thinking along the lines of typing delay, then Off means no typing delay and is also equivalent to 0 on slider.

If I reverse the slider then Off corresponds to 100 on slider 🤔

File: scenes/ui/settings_slider/settings_slider.gd (hunk 2 above)
Comment (TinyTakinTeller) [reply scenes/ui/settings_slider/settings_slider.gd:11]: Hmm, should be possible. 

Would it be better to then only extract the ShakeShaderComponent ? Something like an export variable on the SettingsSlider that selects a target to apply on the internal TitleLabel.

File: scenes/component/control/label/label_typing/label_typing.gd (hunk 3 above)
Comment (mielifica) [reply scenes/component/control/label/label_typing/label_typing.gd:47]: Hmm it does make more sense to see the 0 value as "infinitely fast"

File: scenes/ui/settings_slider/settings_slider.gd (hunk 2 above)
Comment (mielifica) [reply scenes/ui/settings_slider/settings_slider.gd:11]: Also sounds like an option. Is the idea to also extract the typing label functionality into a TypingComponent? (but then what if you want to apply two effects to a title @_@)

Comment (TinyTakinTeller) [reply scenes/ui/settings_slider/settings_slider.gd:11]: Hmm - I'll leave it as is for now. 👀

**Pull request merged**